### PR TITLE
LORE-282 Clean up the spec

### PR DESCRIPTION
**Pull request by kcover**
#### What does this PR do?
*Removes the ErrorMessage field on MetadataTransformations and replaces
  it with a FailureReason and FailureMessage.
*Changes the MetadataType to a String instead of an enum.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
 @brjeter @paouelle @tyler30clemens @austinsteffes @brendan-hofmann 

#### How should this be tested? (List steps with links to updated documentation)
CI
#### Any background context you want to provide?
Corresponds to a PR in the ion-transformation repo

#### What are the relevant tickets?
Fixes #XXXX

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 1]
```diff
@@ -366,11 +371,11 @@ components:
     MetadataType:
       description: The type of metadata.
       type: string
-      enum: [irm]
+      example: irm
     Status:
       description: The status of the operation.
       type: string
-      enum: [In Progress, Done, Failed]
+      enum: [In Progress, Done, Failed, Unknown]
```
Comment (paouelle) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: ❗️ We should not be adding the `Unknown` error at the API level. Unknown in our internal design should either translate in a 503 letting the client know they should retry or a redirect back to the public URL. Both situations are such that the request comes back into our endpoint and hits a newer version of the endpoint that can handle the updated data. The notion of unknown should never leave our bounded context.

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 2]
```diff
@@ -366,11 +371,11 @@ components:
     MetadataType:
       description: The type of metadata.
       type: string
```
Comment (paouelle) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: ❗️  Add a regex for valid URL format
✏️ Might want to update the other URIs on the transform request.

File: models/src/main/java/com/connexta/transformation/rest/models/ErrorMessage.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.List;
+import java.util.Objects;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+/** Error message. */
+@ApiModel(description = "Error message.")
+@javax.annotation.Generated(
+    value = "org.openapitools.codegen.languages.SpringCodegen",
```
Comment (paouelle) [inline models/src/main/java/com/connexta/transformation/rest/models/ErrorMessage.java]: ❓ Should we change those? Same question for all other files

File: models/src/main/java/com/connexta/transformation/rest/models/TransformationPollResponse.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  @Valid
+  public Status getTransformationStatus() {
+    return transformationStatus;
+  }
+
+  public void setTransformationStatus(Status transformationStatus) {
+    this.transformationStatus = transformationStatus;
+  }
+
+  public TransformationPollResponse errorMessage(ErrorMessage errorMessage) {
```
Comment (paouelle) [inline models/src/main/java/com/connexta/transformation/rest/models/TransformationPollResponse.java]: ❗️ This should not be the `ErrorMessage` used as a response body for errors. It should only contain the failure code and message string.

File: models/src/main/java/com/connexta/transformation/rest/models/TransformationPollResponse.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+   */
+  @ApiModelProperty(
+      required = true,
+      value =
+          "An array of the information for each metadata type that will be provided from the transformation request. NOTE this could be empty. ")
+  @NotNull
+  @Valid
+  @Size(min = 0)
+  public List<MetadataInformation> getMetadataInformations() {
+    return metadataInformations;
```
[the remark this answers is not in the document]
Comment (brjeter) [reply models/src/main/java/com/connexta/transformation/rest/models/TransformationPollResponse.java]: Does this really matter? Won't these objects just be created on the client side from the real source of truth which is the HTTP responses?

File: spring/pom.xml [hunk 6]
```diff
@@ -22,33 +22,45 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <name>Transformation :: API :: Java</name>
+    <name>Transformation Rest Spring Stubs</name>
```
Comment (brjeter) [inline spring/pom.xml]:
```suggestion
    <name>Transformation :: API :: Rest Spring Stubs</name>
```

Comment (brjeter) [reply spring/pom.xml]: My thought is that, if the artifact name will have "api" in it then the name of the module should too.

File: models/pom.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>transformation-api</artifactId>
+        <groupId>com.connexta.transformation</groupId>
+        <version>0.3.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <name>Transformation Rest Models</name>
```
Comment (brjeter) [inline models/pom.xml]:
```suggestion
    <name>Transformation :: API :: Rest Models</name>
```
My thought is that, if the artifact name will have "api" in it then the name of the module should too.

**Pull request merged**